// interceptor to catch errors
const errorInterceptor = (error) => {
    // check if it's a server error
    if (!error.response) {
        console.log('Some thing wrongs on server...')
        return Promise.reject(error)
    }

    // all the error responses
    switch (error.response.status) {
        case 400:
            // toast.error(error.response.status, error.message)
            console.log('游니 API | Nothing to display', 'Data Not Found')
            break

        case 401: // authentication error, logout the user
            console.log('游니 API | Please login again', 'Authenticated failed')
            window.sessionStorage.removeItem('user')
            if (error.response.data.EC === -999) {
                console.log('游니 API | Please login again', 'Token Expired')
                window.sessionStorage.setItem("token-expired", true);
                window.location.href = "/login";
            }
            break

        case 403:
            // toast.error(error.response.status, error.message)
            console.log('游니 API | Access denied', 'Data Not Found')
            break

        case 404:
            // toast.error("Some thing wrongs on client...")
            console.log('游니 API | Dataset not found', 'Data Not Found')
            break

        case 422:
            // toast.error(error.response.status, error.message, error.response.data.detail)
            console.log('游니 API | Validation error', 'Unprocessable Content')
            break

        default:
            console.log(error)
    }
    return Promise.reject(error)
}

export default errorInterceptor;